### DISCUSSION TITLE
Raster registration using mutual correspondences in PyMeshLab

### DISCUSSION THREAD

**Discussion by leongn** [cnr-isti-vclab/PyMeshLab#116] [Q&A] [answered]
Hi, 
I've been using MeshLab to texture meshes by using the 'textures>Parametrization + texturing from registered rasters'method. 

In order to align the mesh with the raster I've been using the 'Raster registration using mutual correspondences' method by defining the corresponding points on the raster and the mesh. 

I now wanted to automate this process using PyMeshLab but wasn't able to find the 'Raster registration using mutual correspondences' method. Am I looking in the wrong places or is it not available? And if not, are there any plans to add it to PyMeshLab?

Thanks!

**Comment by alemuntoni** [comment, marked as answer]
Sorry but "Raster registration using mutual correspondences" is a method that belongs to an edit plugin, that's because the main feature of the plugin need manual editing operations that cannot be automatized easily, therefore this method is missing from pymeshlab.

**Reply by leongn** [reply]
That makes sense, thank you for the quick answer.

**Reply by PMDevine1** [reply]
@alemuntoni, how about if you want to import the correspondences from a text file and not do any manual editing operation? Would this be a more feasible implementation in PyMeshLab?

**Reply by alemuntoni** [reply]
It is not at the moment: the codebase of meshlab does not allow to do this. Doing this requires a lot of rewriting (transferring code from an edit to a filter plugin), and I don't think this feature will be available soon in pymeshlab. Sorry about that.

**Comment by misa-rtx** [comment]
Hello, I have created a library in python for texture-mapping similar to Meshlab rasterizer: https://github.com/misa-rtx/PhotoMesh/tree/main

You only need the frames, camera poses/parameters (from COLMAP, for ex), and the reconstructed raw mesh. 
If you run SFM and generate a sparse point cloud, you can use Poisson reconstruction (https://www.open3d.org/docs/latest/tutorial/Advanced/surface_reconstruction.html) and just take the geometry, since the mapped RGB points are low-resolution. 

After getting frames, images.txt and cameras.txt, and poisson.ply, you will be able to easly run the texture mapper allowing high resolution to your mesh :) 

<img width="1924" height="1230" alt="photomesh" src="https://github.com/user-attachments/assets/6efaeed4-c26b-493a-afd0-34246c046cc1" />
